PROTOCOL.md

```diff
@@ -1908,14 +1910,21 @@ The response contains multiple lines:
 
 #### Request Body
 
-The request body should be a JSON string containing the following two optional fields:
+The request body should be a JSON string containing the following optional fields:
  
 - **predicateHints** (type: String, optional): a list of SQL boolean expressions using [a restricted subset of SQL](#sql-expressions-for-filtering), in a JSON array.
   - When it’s present, the server will use the provided predicates as a hint to apply the SQL predicates on the returned files.
     - Filtering files based on the SQL predicates is **BEST EFFORT**. The server may return files that don’t satisfy the predicates.
     - If the server fails to parse one of the SQL predicates, or fails to evaluate it, the server may skip it.
     - Predicate expressions are conjunctive (AND-ed together).
   - When it’s absent, the server will return all of files in the table.
+  - **These will be deprecated once all the client and server implementation move to using jsonPredicateHints below**.
+
+- **jsonPredicateHints** (type: String, optional): query predicates on partition columns specified using [a structured JSON format](#json-predicates-for-filtering), which is encoded using Base64 encoding.
+  - When it’s present, the server will try to use the predicates to filter table's files, which could boost query performance.
+    - As with **predicateHints**, this filtering is **BEST EFFORT**. The server may return files that don’t satisfy the predicates.
+    - If the server encounters any errors during predicate processing (for example, invalid syntax or non existing columns), it will skip filtering and return all the files. 
```

Comment:
Do we need to adjust the content here, though skip filtering, the server may still respect limitHints, which doesn't return all files?

Reply:
I think limitHint today is suppressed if there is predicate filtering.